sale_stock_invoice_snln_enhanced/views/sale_stock_invoice_snln_enhanced.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="sale_stock_report_invoice_snln_document" inherit_id="sale_stock.sale_stock_report_invoice_document">
+        <xpath expr="//table[@name='invoice_snln_table']" position="attributes">
+            <attribute name="style">width: 100%;</attribute>       
+        </xpath>
+
+        <xpath expr="//t[contains(@t-esc,'quantity')]" position="before">
+            <t t-set="decimal_precision" t-value="request.env['decimal.precision'].precision_get('Product Unit of Measure')"/>
+        </xpath>
```

Comment:
Maybe this variable could be set outside the t-foreach loop in order to prevent multiple an unnecessary asignments, e.g. (not tested):
```suggestion
        <xpath expr="//t[@t-set='lot_values')]" position="after">
            <t t-set="decimal_precision" t-value="request.env['decimal.precision'].precision_get('Product Unit of Measure')"/>
        </xpath>
```